core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -198,13 +198,20 @@ func (s *SmartContract) LockAsset(ctx contractapi.TransactionContextInterface, a
 		return "", fmt.Errorf("error in base64 decode of lock information: %+v", err)
 	}
 
-	lockInfoHTLC := &common.AssetLockHTLC{}
-	err = proto.Unmarshal([]byte(lockInfoBytes), lockInfoHTLC)
+	lockInfo := &common.AssetLock{}
+	err = proto.Unmarshal([]byte(lockInfoBytes), lockInfo)
 	if err != nil {
 		return "", logThenErrorf(err.Error())
 	}
+	lockInfoHTLC := &common.AssetLockHTLC{}
+        if lockInfo.LockMechanism == common.LockMechanism_HTLC {
+		err = proto.Unmarshal(lockInfo.LockInfo, lockInfoHTLC)
+		if err != nil {
+			return "", logThenErrorf(err.Error())
+		}
+	}
```

Comment:
Do you think adding an else statement, where we can return some error like "Lock Mechanism not supported", just for safety? I know it is going to be always set a valid value in interop-sdk, but I still I think its a good idea to do that.

Reply:
Yes, it is good to have it here also @sanvenDev . Added now.